### PR TITLE
chore(deps): update dependencies to latest

### PR DESCRIPTION
**Pull request by briandealwis**
Update dependencies in our image:
- Kubectl to 1.20.13
- Helm to 3.7.2
- Kustomize to 4.4.1
- container-structure-test to 1.11.0: I had previously computed a sha512 as the project's shas were incorrect; they're now correct but are sha256
- k3d to 5.2.1
- gcloud to 366.0.0
- bazel to 4.2.2

Also fix an odd path issue: I'm not sure how this happened by the digest paths assumed the Dockerfile was being run from the Skaffold root, whereas `make build_deps` sets the build context to `deploy/skaffold`.

### CODE REVIEW

File: deploy/skaffold/Dockerfile.deps [hunk 1]
```diff
@@ -19,29 +19,29 @@ FROM alpine:3.10 as download-kubectl
 ARG ARCH
 # Track default version installed by Google Cloud SDK: 358.0.0 moved to 1.20(.10)
 # https://cloud.google.com/sdk/docs/release-notes
-ENV KUBECTL_VERSION v1.20.10
+ENV KUBECTL_VERSION v1.20.13
 ENV KUBECTL_URL https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/${ARCH}/kubectl
 # SHAs at gs://kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/${ARCH}/
-COPY deploy/skaffold/digests/kubectl.${ARCH}.sha512 .
```
Comment (tejal29) [inline deploy/skaffold/Dockerfile.deps:25]: Can you double if the cloud build triggers work correctly?

For cloud build job, the workspace is root of skaffold. The last time I checked, `Copy` failed

Comment (tejal29) [reply deploy/skaffold/Dockerfile.deps:25]: Just saw your comment. We changed the release infra to not run `make deps`. Instead the `build_deps` image is now built every single time. 

See -> https://github.com/GoogleContainerTools/skaffold/blob/main/deploy/skaffold/Dockerfile#L16

Maybe we should remove the `make deps` target.

Comment (briandealwis) [reply deploy/skaffold/Dockerfile.deps:25]: Ah, ok.  I'll take this as an opportunity to merge the various deploy/skaffold/Dockerfile.* into a single with independent targets.

**Review by SSGoku369 — COMMENTED** [review]
ok

**Comment by SSGoku369** [comment]
ok

**Comment by tejal29** [comment]
closing this PR. Lets get the deps upgraded as part of v2

**Pull request closed without merging**